contracts/KeysManager.sol

```diff
@@ -357,6 +364,9 @@ contract KeysManager is EternalStorage, IKeysManager {
     function addMiningKey(address _key) public onlyVotingToChangeKeys returns(bool) {
         if (!_withinTotalLimit()) return false;
         if (!initDisabled()) return false;
+        if (hasMiningKeyBeenRemoved(_key)) {
```

Comment:
We could add the same checking into the `swapMiningKey` as well (before calling of `IPoaNetworkConsensus.swapValidatorKey`).